codespell_lib/tests/test_dictionary.py

```diff
@@ -172,18 +173,27 @@ def test_error_checking_in_aspell(err, rep, err_aspell, rep_aspell, match):
 @fname_params
 def test_dictionary_looping(fname, in_aspell):
     """Test that all dictionary entries are valid."""
-    err_dict = dict()
+    file_err_dict = dict()
     with open(fname, 'rb') as fid:
         for line in fid:
             err, rep = line.decode('utf-8').split('->')
             err = err.lower()
-            assert err not in err_dict, 'error %r already exists' % err
+            assert err not in file_err_dict, 'error %r already exists in %s' % (err, fname)
+            # We check if it's in file_err_dict too, so we don't throw two errors when it is
+            assert (err in file_err_dict) or (err not in global_err_dict), 'error %r already exists in another dictionary file' % (err)
             rep = rep.rstrip('\n')
             reps = [r.strip() for r in rep.lower().split(',')]
             reps = [r for r in reps if len(r)]
-            err_dict[err] = reps
-    # check for corrections that are errors (but not self replacements)
-    for err in err_dict:
-        for r in err_dict[err]:
-            assert (r not in err_dict) or (r in err_dict[r]), \
-                ('error %s: correction %s is an error itself' % (err, r))
+            file_err_dict[err] = reps
+            global_err_dict[err] = reps
```

Comment:
... and actually here I would do `global_err_dict[err] = fname`

then below you can say
```
... is an erorr itself in the dictionary file %s' % (err, glrabel_err_dict[err])
```

Reply:
If I do that, I wouldn't be able to catch the case where the correction is the typo, as I wouldn't have that would I? I think the proper solution would be a tuple of filename and corrections wouldn't it?

globel is already in the dictionary :stuck_out_tongue_winking_eye:

Reply:
Do you want me to do anything more with this one?

Reply:
I figured out a decent way to do this -- you do `global_err_dicts = dict()` as you've done here, then do `global_err_dicts[fname] = dict()` at the top of this function, then below when you do it it iterate over:
```
for other_fname, global_err_dict in global_err_dicts.items():
    ...
```
then you have access to all the errors, and immediately can report which `other_fname` to tell the person

But this is PR already fine by me, so only do it if you want to. Do need to get CIs green, though...

Reply:
I've done what I think you meant in https://github.com/codespell-project/codespell/pull/1501/commits/d2488ce042ee9666d735e5c38465c35536fd46b7 although it feels to me like it's just going to check each dictionary individually, unless I'm misunderstanding your intended loops?

Edit, after a bit more pondering I've done https://github.com/codespell-project/codespell/pull/1501/commits/52e9b276a2ac658c21a84ac861216f7fd9c86211 which feels right, just waiting for the CI to find out.

Reply:
Update, this has indeed missed `calculatable`, I think because of the order of my looping, the err and correction need to be added in the right order to catch it as we don't run anything after looping over every dictionary file.